### ISSUE TITLE
update_with_buffer() can cause buffer overflow's

### ISSUE THREAD

**Issue by stianeklund** [emoon/rust_minifb#40]
Filing this as discussed. Issue is that `update_with_buffer()` can cause a overflow and start reading into random heap memory.

This was first discovered on Windows. Will test Linux and update the issue. If it also happens on Linux either it's happening at the OS API level or not.

If possible add checking to `update_with_buffer()` to prevent reading data outside of the set valid range.

**Comment by emoon** [comment]
Thanks! Yeah this makes sense to check. I will try to do this during the weekend it also relates a bit to #38 to make sure it doesn't read outside and such.

**Comment by emoon** [comment]
So I'm thinking about the design for this. Should `update_with_buffer` return a`Result<>` that the user can check or should it return a bool if it failed to update? That is less intrusive or should it just `println!` that it failed. I'm not sure what the best way is here.

Any thoughts?

**Comment by kondrak** [comment]
I think Result<> is the "most Rusty" way to do it if following the guidelines from the book.

**Comment by emoon** [comment]
Agreed. It's a breaking change for anyone using the library but as it's not 1.x+ yet I guess that can be expected.

**Comment by emoon** [comment]
It's not exactly breaking per se I guess (but it will generate a warning)

**Comment by kondrak** [comment]
I think it's worth the change, just bump it to 0.10 :)

**Comment by emoon** [comment]
Yeah will do! Thanks.

**Comment by emoon** [comment]
The change will be now that applications that uses `unwrap` on the result from the `update_with_buffer` call can get this.

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: UpdateFailed("Update failed because input buffer is too small. Required size for 1280 x 724 window (2x scale) is 926720 bytes but the size of the input buffer has the size 921600 bytes")', src/libcore/result.rs:860
```
Hopefully that error should be good enough
